packages/cactus-plugin-ccmodel-hephaestus/src/main/typescript/plugin-ccmodel-hephaestus.ts

```diff
@@ -0,0 +1,911 @@
+    const startTime = new Date();
+    const fnTag = `PluginCcModelHephaestus#constructor()`;
+    if (!options) {
+      throw new Error(`${fnTag} options falsy.`);
+    }
+    Checks.truthy(options.instanceId, `${fnTag} options.instanceId`);
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({
+      label: label,
+      level: level,
+    });
+    this.instanceId = this.options.instanceId;
+    this.crossChainLog = new CrossChainEventLog({
+      name: "HEPHAESTUS_EVENT_LOGS",
+    });
+    this.unmodeledEventLog = new CrossChainEventLog({
+      name: "HEPHAESTUS_UNMODELED_LOGS",
+    });
+    this.nonConformedCrossChainLog = new CrossChainEventLog({
+      name: "HEPHAESTUS_NON_CONFORMANCE_LOGS",
+    });
+
+    this.caseID = "UNDEFINED_CASE_ID";
+
+    this.ethTxObservable = options.ethTxObservable;
+    this.besuTxObservable = options.besuTxObservable;
+    this.fabricTxObservable = options.fabricTxObservable;
+
+    this.sourceLedger = options.sourceLedger;
+    this.targetLedger = options.targetLedger;
+
+    //todo should allow different models to be instantiated
+    this.crossChainModel = new CrossChainModel();
+
+    this.isModeling = true;
+
+    this.nonConformedCCTxs = [];
+
+    const finalTime = new Date();
+    this.log.debug(
+      `EVAL-${this.className}-SETUP-CONSTRUCTOR:${finalTime.getTime() - startTime.getTime()}`,
+    );
+  }
+
+  getOpenApiSpec(): unknown {
+    throw new Error("Method not implemented.");
+  }
+
+  get ccModel(): CrossChainModel {
+    return this.crossChainModel;
+  }
+
+  get numberEventsLog(): number {
+    return this.crossChainLog.numberEvents();
+  }
+
+  get numberEventsUnmodeledLog(): number {
+    return this.unmodeledEventLog.numberEvents();
+  }
+
+  public purgeNonConformedEvents(): void {
+    this.nonConformedCrossChainLog.purgeLogs();
+  }
+
+  get numberEventsNonConformedLog(): number {
+    return this.nonConformedCrossChainLog.numberEvents();
+  }
+
+  public purgeCrossChainEvents(): void {
+    this.crossChainLog.purgeLogs();
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public getCaseId(): string {
+    return this.caseID;
+  }
+
+  public setIsModeling(bool: boolean): void {
+    this.isModeling = bool;
+  }
+
+  public setCaseId(id: string): void {
+    this.unmodeledEventLog.purgeLogs();
+    this.caseID = id;
+  }
+
+  public async onPluginInit(): Promise<unknown> {
+    return;
+  }
+
+  public async shutdown(): Promise<void> {
+    this.log.info(`Shutting down...`);
+    const serverMaybe = this.getHttpServer();
+    if (serverMaybe.isPresent()) {
+      this.log.info(`Awaiting server.close() ...`);
+      const server = serverMaybe.get();
+      await promisify(server.close.bind(server))();
+      this.log.info(`server.close() OK`);
+    } else {
+      this.log.info(`No HTTP server found, skipping...`);
+    }
+  }
+
```

Comment:
@RafaelAPB This is no longer a needed method of the plugin interfaces and you can also delete the `getHttpServer` method and the `httpServer` field itself. Makes no difference but I figured it helps to tidy up a little.

Reply:
if i remove the `shutdown()` method i get an error saying `Property 'shutdown' is missing in type 'CcModelHephaestus' but required in type 'IPluginWebService'`, should i keep it with just the log message then?